tests/python/unittest/test_operator.py

```diff
@@ -27,6 +27,83 @@
 from common import setup_module, with_seed
 import unittest
 
+def check_lstm_with_type(xpu, type1, type2, atol):
+    X = mx.sym.Variable('x')
+    Params = mx.sym.Variable('params')
+    HX = mx.sym.Variable('state')
+    CX = mx.sym.Variable('state_cell')
+    T, N, I, H, nd, nl = 4, 16, 800, 800, 1, 1
+    size = (I + H + 2) * H * 4 * nd;                # first layer
+    x1 = mx.random.uniform(-1, 1, (T, N, I), ctx=xpu, dtype=type1)
+    wx = mx.random.uniform(-1, 1, (4 * H, I), ctx=xpu,dtype=type1)
+    wh = mx.random.uniform(-1, 1, (4 * H, H), ctx=xpu,dtype=type1)
+    bx = mx.nd.zeros((4 * H,), ctx=xpu, dtype=type1)
+    bh = mx.nd.zeros((4 * H,), ctx=xpu, dtype=type1)
+    x1.attach_grad()
```

Comment:
why do you need to manually attach grad??

Reply:
attach_grad is used to create gradient buffer for these NDArrays here. Do you mean this can be implemented in other ways or do you have any suggestion about this piece of code?

Reply:
In case use stateful OP, what's your opinion @eric-haibin-lin ?